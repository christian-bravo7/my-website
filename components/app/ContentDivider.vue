<template>
  <section
    class="content-divider"
  >
    <div
      class="content-divider__section"
      :class="{ 'content-divider__section--hidden': second }"
    />
    <div
      class="content-divider__section"
      :class="{ 'content-divider__section--hidden': first }"
    />
  </section>
</template>

<script lang="ts">
import { Vue, Component, Prop } from 'nuxt-property-decorator';

@Component
export default class ContentDivider extends Vue {
  @Prop({ type: Boolean, default: false })
  readonly first!: boolean;

  @Prop({ type: Boolean, default: false })
  readonly second!: boolean;
}
</script>

<style lang="scss" scoped>
.content-divider {
  display: grid;
  grid-template-columns: 1fr 1fr;
  height: rem(80);

  &__section {
    height: 100%;
    background-color: var(--text-light);

    &--hidden {
      opacity: 0;
    }
  }
}

@include mobile {
  .content-divider {
    height: rem(40);
  }
}
</style>
