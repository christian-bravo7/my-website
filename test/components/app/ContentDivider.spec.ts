import ContentDivider from '@/components/app/ContentDivider.vue';
import { mount } from '@vue/test-utils';

describe('ContentDivider component', () => {
  it('should render correclty', () => {
    const wrapper = mount(ContentDivider);

    expect(wrapper.html()).toMatchSnapshot();
  });
});
