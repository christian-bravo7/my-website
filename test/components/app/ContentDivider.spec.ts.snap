// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentDivider component should render correclty 1`] = `
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 1440 320" class="text-gray-200 dark:text-blue-900">
  <path fill="currentColor" fill-opacity="1" d="M0,192L360,160L720,96L1080,224L1440,224L1440,320L1080,320L720,320L360,320L0,320Z"></path>
</svg>
`;
